lib/src/command/global_run.dart

```diff
@@ -57,8 +61,16 @@ class GlobalRunCommand extends PubCommand {
       log.warning('The --mode flag is deprecated and has no effect.');
     }
 
-    var exitCode = await globals.runExecutable(package, executable, args,
-        enableAsserts: argResults['enable-asserts'] || argResults['checked']);
+    final experiments = argResults['enable-experiment'] as List;
+    final vmArgs = <String>[
+      if (experiments.isNotEmpty) "--enable_experiment=${experiments.join(',')}"
```

Comment:
Silly nit: use `enable-experiment` for the SDK. It supports both `_` and `-` for separators, but we should be consistant!